### PR TITLE
check that system allows to pass traffic via default and vlan flows when they are on the same port

### PR DESCRIPTION
**Pull request by andriidovhan**
check that system allows to create both full port and port+vlan flows on the same port

Close: #2433

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 1]
```diff
@@ -642,6 +642,68 @@ class FlowCrudSpec extends BaseSpecification {
...
+        flowVlanPort.allocateProtectedPath = true
+        flowHelper.addFlow(flowVlanPort)
+
+        and: "Create the same flow with full port"
+        def flowFullPort = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowFullPort.maximumBandwidth = bandwidth
+        flowFullPort.source.vlanId = 0
+        flowFullPort.destination.vlanId = 0
+        flowFullPort.allocateProtectedPath = true
+        flowHelper.addFlow(flowFullPort)
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: You should use the same ports on src and dst. Current test is not relevant

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 2]
```diff
@@ -642,6 +642,68 @@ class FlowCrudSpec extends BaseSpecification {
...
+        then: "The flow with full port has less priority than flow with vlan port"
+        def flowVlanPortInfo = database.getFlow(flowVlanPort.id)
+        def flowFullPortInfo = database.getFlow(flowFullPort.id)
+
+        def rules = [srcSwitch.dpId, dstSwitch.dpId].collectEntries {
+            [(it): northbound.getSwitchRules(it).flowEntries]
+        }
+
+        // can't be imported safely org.openkilda.floodlight.switchmanager.SwitchManager.FLOW_FULL_PORT_PRIORITY
+        // [ERROR] General error during semantic analysis: java.lang.NoClassDefFoundError:
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: the info is not valuable. We know why that happens, why insert error text?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: You may be mistaken.
Here we create flows with traffgen ports.
```
FlowCreatePayload randomFlow(Switch srcSwitch, Switch dstSwitch, boolean useTraffgenPorts = true,
                                 List<FlowPayload> existingFlows = []) {
```
As a result this test alsways gets the same port for both flows

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: text of error was removed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 1 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: agree

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 3]
```diff
@@ -625,6 +704,102 @@ class FlowCrudSpec extends BaseSpecification {
...
+            direction.setResources(resources)
+            assert traffExam.waitExam(direction).hasTraffic()
+        }
+
+        and: "Cleanup: Delete the flows"
+        [vlanFlow, defaultFlow].each { flow -> flowHelper.deleteFlow(flow.id) }
+    }
+
+    def "Unable to create two default flow on the same port"() {
+        given: "At least 3 traffGen switches"
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: I mean that three traffGen switches are available in our topology

[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: sorry, I got your point

[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: fixed

**Pull request merged**